internal/xds/translator/ratelimit.go

```diff
@@ -329,20 +329,12 @@ func BuildRateLimitServiceConfig(irListener *ir.HTTPListener) *rlsconfv3.RateLim
 func buildRateLimitServiceDescriptors(global *ir.GlobalRateLimit) []*rlsconfv3.RateLimitDescriptor {
```

Comment:
Nit: This piece of code is hard to understand. Add an example to the comment will help later readers to understand it.

input:
```
name: "first-listener"
...
routes:
- name: "first-route"
  traffic:
    rateLimit:
      global:
        rules:
        - headerMatches:
          - name: "x-org
            exact: "foo"
          - name: "x-use
            exact: "bar"
          cidrMatch:
            cidr: 192.16
            maskLen: 16
          limit:
            requests: 5
            unit: second
```

output:
```
name: first-listener
domain: first-listener
descriptors:
  - key: first-route
    value: first-route
    rate_limit: null
    descriptors:
      - key: rule-0-match-0
        value: rule-0-match-0
        rate_limit: null
        descriptors:
          - key: rule-0-match-1
            value: rule-0-match-1
            rate_limit: null
            descriptors:
              - key: masked_remote_address
                value: 192.168.0.0/16
                rate_limit:
                  requests_per_unit: 5
                  unit: SECOND
                  unlimited: false
                  name: ""
                  replaces: []
                descriptors: []
                shadow_mode: false
                detailed_metric: false
            shadow_mode: false
            detailed_metric: false
        shadow_mode: false
        detailed_metric: false
    shadow_mode: false
    detailed_metric: false
```

Reply:
I've added some natural language to describe this function, imo adding example code is unnecessary, since it has the same content as the test does.